File: src/components/dialog/error_dialog.py
# Standard library
from typing import Callable

# Third-Party libraries
from flet import CrossAxisAlignment, Page, icons

# Create widgets
from ..widgets import Dialog, ElevatedButton

# Utils
from utils import check_type


class ErrorDialog(Dialog):
    @check_type
    def __init__(self, app_page: Page, overlay: Callable):
        self.app_page = app_page
        self.overlay = overlay

        self.button_close_dialog = ElevatedButton(
            text_button="Ok", function=lambda e: self.change_state(app_page)
        )

        super().__init__(
            icon=True,
            title=icons.ERROR,
            title_size=1.3,
            content="",
            content_size=23,
            actions=[self.button_close_dialog],
            actions_alignment=CrossAxisAlignment.END,
            overlay=self.overlay,
            app_page=self.app_page,
        )

    @check_type
    def show(self, error: str):
        self.update_content(error)

        return super().show()

File: lib/components/dialogs/error_dialog.dart
import 'package:flutter/material.dart' show BuildContext, Navigator;

import '../widgets/text_button.dart' show CreateTextButton;
import 'base_dialog.dart' show BaseDialog;

class ErrorDialog extends BaseDialog {
  const ErrorDialog({
    super.key,
    required super.actions,
    required super.title,
    super.content,
  });

  static Future<T?> show<T>(BuildContext context, {required String content}) {
    return BaseDialog.show<T>(
      context,
      title: 'Error',
      content: content,
      actions: [
        CreateTextButton(
          text: 'Ok',
          onPressed: () => Navigator.of(context).pop(true),
        ),
      ],
    );
  }
}
